docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -37,6 +37,8 @@ When compiling code in any of the situations in the below table, add the corresp
 | for shared libraries | `-fPIC -shared`                |
 | for x86_64           | `-fcf-protection=full`         |
 | for aarch64          | `-mbranch-protection=standard` |
+| for instrumented test code | `-ftrivial-auto-var-init=pattern` |
```

Comment:
As`-ftrivial-auto-var-init` interferes with unitialized memory tracking in tools such as Valgrind and AddressSanitizer I don't. think it should be enabled for instrumented test code. There was a discussion on this in [this comment to #245](https://github.com/ossf/wg-best-practices-os-developers/issues/245#issuecomment-1785479519).

Reply:
@thesamesam Thanks, I was thinking of MSAN too, not ASAN. The `-ftrivial-auto-var-init` does interfere with MSAN as can be seen here: https://godbolt.org/z/6qTPz9n6h

Reply:
If you don't use MSAN, is this still best to not use? I think many developers don't use MSAN.

Reply:
It'd be easy to remove the `=pattern` case. That's probably the best strategy, at least for now.

Reply:
Ok, let's do that. We can discuss =pattern in the details, without recommending its use in particular.